### ISSUE TITLE
training

### ISSUE THREAD

**Issue by junkangfang** [AlanJiang98/EvRGBHand#4]
I would like to ask if your code supports multi-card training？

**Comment by AlanJiang98** [comment]
Sure!
Our model is trained on multi-GPUs.
In https://github.com/AlanJiang98/EvRGBHand/blob/main/src/configs/config/evrgbhand.yaml, set config['exper']['distributed'] to True will enable the training. 
The code in https://github.com/AlanJiang98/EvRGBHand/blob/main/train.py#L81 shows the setting of training.

We haven't employ multi-node training since the training load is not really heavy.

**Comment by junkangfang** [comment]
Thanks for your answer! ! When I was training on a single card, I found that the early loss was always around 200. Problems would occur when using torchrun for multi-card training, and the loss was still 200 at 50 epochs.
![image](https://github.com/user-attachments/assets/032ffcb9-c2b4-461e-b45c-7d3fd811c1c6)

**Comment by AlanJiang98** [comment]
The issue seems to be the input tensor shape error of the model. Please check the input data dimension.

**Comment by junkangfang** [comment]
This problem did not occur when I was training with a single card. And I found that the input dimensions are correct. Such a strange loss makes me very confused.

**Comment by junkangfang** [comment]
And there was no problem when I tested using your pre-trained model.

**Comment by junkangfang** [comment]
Sorry to bother you again. I tried several methods you gave, but the results of reproduction are somewhat different from the paper. I completely use your environment, but I don’t know where the problem is. I am very interested in your work. If you have time to answer, I will be very grateful！！！

**Comment by AlanJiang98** [comment]
When submitting to CVPR, the numerical results from our multiple experiments showed some fluctuation but without significant differences. We suggest you check if the training epochs and learning rate settings are appropriate, and whether the input tensor dimension issue you encountered before has been resolved. 
You can provide some details about the training process and results, so we can analyze the issue.

**Comment by junkangfang** [comment]
![image](https://github.com/user-attachments/assets/ed27b338-59e5-4664-ab75-935fe973afbc)
I used 3 cards, the learning rate was 0.0002, and I used contiguous() to solve the gradient problem. I will consult you again when the experimental results come out tomorrow. Thank you! !

**Comment by junkangfang** [comment]
Hello, I would like to ask you some questions. When I run the program, there are 24 training data and 8 test data, but I found that the training and test data on the right hand side of the data set are 30 and 12 respectively. Is there something wrong with my data set? My training loss is normal, but the test result is still not ideal.

![image](https://github.com/user-attachments/assets/b3648200-ea70-4794-8df2-857345f85cd8)
![image](https://github.com/user-attachments/assets/26763a0e-9cb9-4f59-b000-d6cae828840a)

**Comment by zxyemo** [comment]
Our method is trained solely on data from normal scenarios. We simulate data for extreme scenarios via degradation. You can modify the `get_events_annotations_per_sequence` function in `src/datasets/EvRealHands.py` to utilize the full dataset, encompassing all scenarios, for training. 
To improve model performance, we generated a synthetic dataset from the InterHand2.6M dataset using the v2e event simulator. We then trained our model using both this synthetic dataset and the EvRealHands dataset. I don't recall the specific loss during the training process, you can check whether overfitting is the reason for the performance drop.

**Comment by junkangfang** [comment]
> Our method is trained solely on data from normal scenarios. We simulate data for extreme scenarios via degradation. You can modify the `get_events_annotations_per_sequence` function in `src/datasets/EvRealHands.py` to utilize the full dataset, encompassing all scenarios, for training. To improve model performance, we generated a synthetic dataset from the InterHand2.6M dataset using the v2e event simulator. We then trained our model using both this synthetic dataset and the EvRealHands dataset. I don't recall the specific loss during the training process, you can check whether overfitting is the reason for the performance drop.

I was wondering if you would make public the synthetic dataset you generated from the InterHand2.6M dataset using the v2e event simulator?

**Comment by zxyemo** [comment]
Due to the large size of the synthetic dataset, we don't have plans to publicly release it at this time. You can easily use the v2e event simulator to generate your own.

**Comment by junkangfang** [comment]
Thanks for your answer, I will close this question.

**Comment by 411ya** [comment]
I would like to ask how to solve the gradient problem. Early loss was always around 200 and was still 200 at 100 epochs. But in another training, the loss was 9 at 50 epochs.

**Comment by LucasForeverForward** [comment]
> ![image](https://github.com/user-attachments/assets/ed27b338-59e5-4664-ab75-935fe973afbc) I used 3 cards, the learning rate was 0.0002, and I used contiguous() to solve the gradient problem. I will consult you again when the experimental results come out tomorrow. Thank you! !

Hello, sorry to bother you. I also noticed that the loss stays around 200 during training. May I ask how you solved this problem?

**Comment by LucasForeverForward** [comment]
> I would like to ask how to solve the gradient problem. Early loss was always around 200 and was still 200 at 100 epochs. But in another training, the loss was 9 at 50 epochs.

Hello, sorry to bother you. I also noticed that the loss stays around 200 during training. I wonder if you have solved this problem.

**Comment by menghang17** [comment]
> 由于合成数据集体积庞大，我们目前没有公开发布的计划。你可以轻松地用v2e事件模拟器生成自己的事件。

您好，抱歉打扰你。相关参数可以公开下吗，以及annot/InterHand2.6M_train_MANO_NeuralAnnot_new.json需要从哪里获取，Interhand官网获取得到的，训练显示mano['root_joint']不存在，难道要置空？
